Clean up the Kotlin DSL Primer in the User Manual (#31812)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/authoring-builds/basics/plugins.adoc

The <<#sec:plugins_block,plugins DSL>> is recommended to resolve and apply plugins in one step.

[[sec:binary_plugin_locations]]

> == Applying plugins

Reviewer: 🤔  I noticed that the first couple snippets this section is Groovy specific. Since we are making changes here, let's multi-DSL this snippet as well.

I mean this one:
```
plugins {
    id «plugin name»
}
```

And the follow up with the version.

---

Looking at them now, it seems there is little value in having the pseudo-code samples. They are simple enough, users would be able to extrapolate from the concrete samples that we, in turn, can make testable.

Author: I left the pseudo snippets as-is but I added examples below in both languages.

}
----

For example:

====
[.multi-language-sample]
=====
.build.gradle.kts
[source, kotlin]
----
plugins {

>     id("com.gradleup.shadow") version "8.3.4"

Reviewer 2: Would it make sense to also change this to a regular member function call rather than the inxix one? I mean `id("com.gradleup.shadow").version("8.3.4")`.

We have not made the decision to deprecate the infix notation here in Kotlin DSL, but the regular member call could potentially be easier to migrate to DCL in the future.

@eskatos WDYT?

Reviewer 3: Let's address this on its own, outside this PR

You can also encapsulate the versions of external plugins by composing the build logic using your own <<sharing_build_logic_between_subprojects#sec:sharing_logic_via_convention_plugins,convention plugins>>.

[[sec:root_plugins_dsl]]

> === 3. Applying plugins from the project root

Reviewer: ```suggestion
=== 3. Applying plugins declared in the root project
```

I think it's important to be consistent with the terminology, though, I am not sure whether this terminology is the right one to use.

We can use "declared plugin" in the context of this page/section as the plugin that was apply-falsed. Maybe we should find a better word than "declared", but importantly it's not the same as "applied". I see that some other places in the text refer to this as "made available".

"Project root" is not a thing, or not a thing in this context, it should be "root project".

But also consider making this section more generic, talking about any parent project. Because the plugins don't have to be declared in the root project to be applied in the childred.

Author: Fixed. Thank you for the suggestion.

> You can apply plugins from the root project in a multi-project build to configure common logic and behavior across all subprojects. The root project is the project at the top of the directory hierarchy.

Reviewer: > across all subprojects

This seems misleading. When you apply a plugin, which was declared in a parent/root project, that plugin is only applied to the current project. It does not get transitively applied to all subprojects.

```suggestion
You can apply plugins from the root project in a multi-project build to share common logic and behavior with other projects. The root project is the project at the top of the directory hierarchy.
```

Author: Fixed.
